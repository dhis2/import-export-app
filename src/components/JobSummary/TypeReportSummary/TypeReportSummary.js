import PropTypes from 'prop-types'
import React from 'react'
import { statsPropType, messagesPropType } from '../helper.js'
import { Messages } from '../Messages/Messages.js'
import { SingleSummary } from '../SingleSummary/SingleSummary.js'
import { TypeCount, typeCountStatsPropTypeObj } from '../TypeCount/TypeCount.js'

const TypeReportSummary = ({ overviewStats, stats, messages }) => {
    return (
        <div>
            <SingleSummary importCount={overviewStats} />
            <TypeCount stats={stats} />
            <Messages messages={messages} />
        </div>
    )
}

TypeReportSummary.propTypes = {
    messages: messagesPropType.isRequired,
    overviewStats: statsPropType.isRequired,
    stats: PropTypes.arrayOf(PropTypes.exact(typeCountStatsPropTypeObj))
        .isRequired,
}

export { TypeReportSummary }
