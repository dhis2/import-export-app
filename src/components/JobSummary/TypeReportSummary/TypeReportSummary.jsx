import PropTypes from 'prop-types'
import React from 'react'
import { statsPropType, messagesPropType } from '../helper.js'
import { Messages } from '../Messages/Messages.jsx'
import { SingleSummary } from '../SingleSummary/SingleSummary.jsx'
import { TypeCount, typeCountStatsPropTypeObj } from '../TypeCount/TypeCount.jsx'

const TypeReportSummary = ({ overviewStats, stats, messages }) => {
    return (
        <div>
            <SingleSummary importCount={overviewStats} />
            <TypeCount stats={stats} />
            <Messages messages={messages} />
        </div>
    )
}

TypeReportSummary.propTypes = {
    messages: messagesPropType.isRequired,
    overviewStats: statsPropType.isRequired,
    stats: PropTypes.arrayOf(PropTypes.exact(typeCountStatsPropTypeObj))
        .isRequired,
}

export { TypeReportSummary }
