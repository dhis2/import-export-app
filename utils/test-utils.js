import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
