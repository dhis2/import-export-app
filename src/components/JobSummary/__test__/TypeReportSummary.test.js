import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { TypeReportSummary } from '../TypeReportSummary/TypeReportSummary.js'

const props = {
    overviewStats: {
        imported: 1,
        deleted: 2,
        ignored: 3,
        updated: 4,
        total: 10,
    },
    stats: [
        {
            type: 'DefaultGmlImportService',
            created: 1,
            deleted: 2,
            ignored: 3,
            updated: 4,
            total: 10,
        },
        {
            type: 'SndGmlImportService',
            created: 1,
            deleted: 2,
            ignored: 3,
            updated: 4,
            total: 10,
        },
    ],
    messages: [
        {
            uid: 'abc',
            type: 'DefaultGmlImportService',
            property: 'def',
            message: 'message one',
        },
        {
            uid: 'xyz',
            type: 'SndGmlImportService',
            property: 'def',
            message: 'message one',
        },
    ],
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<TypeReportSummary {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
