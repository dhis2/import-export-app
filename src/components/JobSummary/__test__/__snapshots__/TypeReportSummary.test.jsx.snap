// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="container"
        data-test="job-summary-single-summary"
      >
        <span
          class="label"
        >
          Summary
        </span>
        <table
          class="jsx-2430604489 "
          data-test="dhis2-uicore-table"
        >
          <thead
            data-test="dhis2-uicore-tablehead"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerowhead"
            >
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Created
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Deleted
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Ignored
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Updated
              </th>
              <th
                class="jsx-1796049005 "
                data-test="dhis2-uicore-tablecellhead"
              >
                Total
              </th>
            </tr>
          </thead>
          <tbody
            data-test="dhis2-uicore-tablebody"
          >
            <tr
              class="jsx-3124144591 zebraStriping"
              data-test="dhis2-uicore-tablerow"
            >
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                1
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                2
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                3
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                4
              </td>
              <td
                class="jsx-996330601 "
                data-test="dhis2-uicore-tablecell"
              >
                10
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="container"
      data-test="job-summary-type-count"
    >
      <span
        class="label"
      >
        Details by type
      </span>
      <table
        class="jsx-2430604489 "
        data-test="dhis2-uicore-table"
      >
        <thead
          data-test="dhis2-uicore-tablehead"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerowhead"
          >
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Type
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Created
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Deleted
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Ignored
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Updated
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Total
            </th>
          </tr>
        </thead>
        <tbody
          data-test="dhis2-uicore-tablebody"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              DefaultGmlImportService
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              1
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              2
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              3
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              4
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              10
            </td>
          </tr>
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              SndGmlImportService
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              1
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              2
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              3
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              4
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              10
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="container"
      data-test="job-summary-messages"
    >
      <span
        class="label"
      >
        Messages
      </span>
      <table
        class="jsx-2430604489 "
        data-test="dhis2-uicore-table"
      >
        <thead
          data-test="dhis2-uicore-tablehead"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerowhead"
          >
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              UID
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Type
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Property
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Message
            </th>
          </tr>
        </thead>
        <tbody
          data-test="dhis2-uicore-tablebody"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              abc
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              DefaultGmlImportService
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              def
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              message one
            </td>
          </tr>
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              xyz
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              SndGmlImportService
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              def
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              message one
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
